source/client/remote_process_impl.cc

```diff
@@ -0,0 +1,58 @@
+#include "client/remote_process_impl.h"
+
+#include <grpc++/grpc++.h>
+
+#include <memory>
+
+#include "nighthawk/client/output_collector.h"
+
+#include "api/client/options.pb.h"
+#include "api/client/output.pb.h"
+
+#include "common/uri_impl.h"
+
+#include "client/options_impl.h"
+
+namespace Nighthawk {
+namespace Client {
+
+RemoteProcessImpl::RemoteProcessImpl(const Options& options,
+                                     nighthawk::client::NighthawkService::Stub& stub)
+    : options_(options), stub_(stub) {}
+
+bool RemoteProcessImpl::run(OutputCollector& collector) {
+  nighthawk::client::ExecutionRequest request;
+  nighthawk::client::ExecutionResponse response;
+  grpc::ClientContext context;
+  auto r = stub_.ExecutionStream(&context);
+
+  *request.mutable_start_request()->mutable_options() = *options_.toCommandLineOptions();
+  // We don't forward the option that requests remote execution. Today,
+  // nighthawk_service will ignore the option, but if someone ever changes that this
+  // is probably desireable.
+  request.mutable_start_request()->mutable_options()->mutable_nighthawk_service()->Clear();
+
+  if (r->Write(request, {}) && r->Read(&response)) {
+    if (response.has_output()) {
+      collector.setOutput(response.output());
+    } else {
+      ENVOY_LOG(error, "Remote execution failed");
```

Comment:
Should this message be closer to "remote execution provided no output"? Or is the lack of output a clear enough signal of a failure?

Reply:
This is a clear signal of a failure; the service will always provide output (with at least some counters indicating connection pool failures).